zipline/finance/order.py

```diff
@@ -0,0 +1,257 @@
+            if event.price >= self.stop:
+                stop_reached = True
+        elif order_type == SELL | STOP:
+            if event.price <= self.stop:
+                stop_reached = True
+        elif order_type == BUY | LIMIT:
+            if event.price <= self.limit:
+                limit_reached = True
+        elif order_type == SELL | LIMIT:
+            # This is a SELL LIMIT order
+            if event.price >= self.limit:
+                limit_reached = True
+
+        return (stop_reached, limit_reached, sl_stop_reached)
+
+    def handle_split(self, split_event):
+        ratio = split_event.ratio
+
+        # update the amount, limit_price, and stop_price
+        # by the split's ratio
+
+        # info here: http://finra.complinet.com/en/display/display_plain.html?
+        # rbid=2403&element_id=8950&record_id=12208&print=1
+
+        # new_share_amount = old_share_amount / ratio
+        # new_price = old_price * ratio
+
+        self.amount = int(self.amount / ratio)
+
+        if self.limit is not None:
+            self.limit = round(self.limit * ratio, 2)
+
+        if self.stop is not None:
+            self.stop = round(self.stop * ratio, 2)
+
+    @property
+    def status(self):
+        if not self.open_amount:
+            return ORDER_STATUS.FILLED
+        elif self._status == ORDER_STATUS.HELD and self.filled:
+            return ORDER_STATUS.OPEN
+        else:
+            return self._status
+
+    @status.setter
+    def status(self, status):
+        self._status = status
+
+    def cancel(self):
+        self.status = ORDER_STATUS.CANCELLED
+
+    def reject(self, reason=''):
+        self.status = ORDER_STATUS.REJECTED
+        self.reason = reason
+
+    def hold(self, reason=''):
+        self.status = ORDER_STATUS.HELD
+        self.reason = reason
+
+    @property
+    def open(self):
+        return self.status in [ORDER_STATUS.OPEN, ORDER_STATUS.HELD]
+
+    @property
+    def triggered(self):
+        """
+        For a market order, True.
+        For a stop order, True IFF stop_reached.
+        For a limit order, True IFF limit_reached.
+        """
+        if self.stop is not None and not self.stop_reached:
+            return False
+
+        if self.limit is not None and not self.limit_reached:
+            return False
+
+        return True
+
+    @property
+    def open_amount(self):
+        return self.amount - self.filled
+
+    def __repr__(self):
+        """
+        String representation for this object.
+        """
+        return "Order(%s)" % self.to_dict().__repr__()
+
+    def __unicode__(self):
+        """
+        Unicode representation for this object.
+        """
+        return text_type(repr(self))
+
+    def __getstate__(self):
+
+        state_dict = \
+            {k: v for k, v in iteritems(self.__dict__)
+                if not k.startswith('_')}
+
+        state_dict['_status'] = self._status
+
+        STATE_VERSION = 1
```

Comment:
@richafrank here is the skip we discussed about the version tests not being run, https://github.com/quantopian/zipline/commit/82affb639f24bbd8a9509c33c130088798f2b1b7